src/ducks/upload/UploadQueue.jsx

```diff
@@ -38,6 +38,33 @@ const Item = translate()(({ t, file, status }) => {
   )
 })
 
+const Header = translate()(({ t, total, done, successful, onDoubleClick, onClose }) => (
+  <h4 className={styles['upload-queue-header']} onDoubleClick={onDoubleClick}>
+    {
+      done < total &&
+      <div className={styles['upload-queue-header-inner']}>
+        <span className='coz-desktop'>
+          {t('UploadQueue.header', { smart_count: total })}
+        </span>
+        <span className='coz-mobile'>
+          {t('UploadQueue.header_mobile', { done, total })}
+        </span>
+      </div>
+    }
+    {
+      done >= total &&
+      <div className={styles['upload-queue-header-inner']}>
+        <span>
+          {t('UploadQueue.header_done', { done: successful, total })}
+        </span>
+        <button className={classNames(styles['btn-close'])} onClick={onClose}>
+          { t('UploadQueue.close') }
+        </button>
+      </div>
+    }
+  </h4>
+))
```

Comment:
I understand. It's a fair point, I wasn't very happy with this code either but wasn't sure how to make it better. Let's go with that last approach, otherwise we'll forget about this.